groovy-language-server/src/test/java/com/palantir/ls/groovy/GroovyLanguageServerTest.java

```diff
@@ -60,6 +63,62 @@ public CompilerWrapper get() {
+        assertTrue(result.getCapabilities().isDocumentSymbolProvider());
+        assertTrue(result.getCapabilities().isWorkspaceSymbolProvider());
+    }
+
+    @Test
+    public void testInitialize_relativeWorkspacePath() throws InterruptedException, ExecutionException, IOException {
+        File workspaceRoot = Paths.get("").toAbsolutePath().resolve("something").toFile();
+        // Create a directory in our working directory
+        assertTrue(workspaceRoot.mkdir());
+
+        GroovyLanguageServer server =
+                new GroovyLanguageServer(new CompilerWrapperProvider() {
+                    @Override
+                    public void set(CompilerWrapper compilerWrapper) {}
+
+                    @Override
+                    public CompilerWrapper get() {
+                        return Mockito.mock(CompilerWrapper.class);
+                    }
+                }, Mockito.mock(LanguageServerConfig.class), Mockito.mock(TextDocumentService.class),
+                        Mockito.mock(WorkspaceService.class), Mockito.mock(WindowService.class));
+        InitializeParams params =
+                new InitializeParamsBuilder().capabilities(new ClientCapabilitiesImpl()).processId(1)
+                        .rootPath("something").build();
+        InitializeResult result = server.initialize(params).get();
+        assertThat(server.getWorkspaceRoot(), is(workspaceRoot.toPath()));
+        assertThat(result.getCapabilities().getTextDocumentSync(), is(TextDocumentSyncKind.Incremental));
+        assertTrue(result.getCapabilities().isDocumentSymbolProvider());
+        assertTrue(result.getCapabilities().isWorkspaceSymbolProvider());
+
+        // Test normalization
+        params = new InitializeParamsBuilder().capabilities(new ClientCapabilitiesImpl()).processId(1)
+                        .rootPath("./something").build();
+        result = server.initialize(params).get();
+        assertThat(server.getWorkspaceRoot(), is(workspaceRoot.toPath()));
+        assertThat(result.getCapabilities().getTextDocumentSync(), is(TextDocumentSyncKind.Incremental));
+        assertTrue(result.getCapabilities().isDocumentSymbolProvider());
+        assertTrue(result.getCapabilities().isWorkspaceSymbolProvider());
+
+        params = new InitializeParamsBuilder().capabilities(new ClientCapabilitiesImpl()).processId(1)
+                        .rootPath("somethingelse/../something/../something").build();
+        result = server.initialize(params).get();
+        assertThat(server.getWorkspaceRoot(), is(workspaceRoot.toPath()));
+        assertThat(result.getCapabilities().getTextDocumentSync(), is(TextDocumentSyncKind.Incremental));
+        assertTrue(result.getCapabilities().isDocumentSymbolProvider());
+        assertTrue(result.getCapabilities().isWorkspaceSymbolProvider());
+
+        // Delete the directory we created in our working directory
+        assertTrue(workspaceRoot.delete());
```

Comment:
We're actually creating a file in our working directory, not in a temp directory.